debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/ModularityHelper.java

```diff
@@ -0,0 +1,315 @@
+				while (slotIterator.hasNext()) {
+					J9ModulePointer modulePtr = slotIterator.next();
+					if (filter.filter(modulePtr, filterArg)) {
+						outputter.print(modulePtr, out);
+					}
+				}
+			}
+		}
+	}
+
+
+	/**
+	 * Traverses through all loaded modules.
+	 * Uses outtputter to print details about modules
+	 * matched by filter.
+	 * 
+	 * @param    out        The printstream that will be provided to outputter
+	 * @param    filter     Used to determine whether a module will be output.
+	 *                      Will be run on all loaded modules.
+	 * @param   outputter   Used to output details about a module that was matched
+	 *                      by the filter. Will be run on any module that filter
+	 *                      returns true for.
+	 * @param   filterArg   A string that is to be used as additional data for filter.
+	 *                      Will be passed to filter everytime filter is called.
+	 */
+	public static void iteratePackages(PrintStream out, PackageIteratorFilter filter, PackageOutput outputter, String filterArg) throws CorruptDataException {
+		J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+		if (JavaVersionHelper.ensureJava9AndUp(vm, out)) {
+			GCClassLoaderIterator iterator = GCClassLoaderIterator.from();
+			while (iterator.hasNext()) {
+				J9ClassLoaderPointer classLoaderPointer = iterator.next();
+				HashTable<J9PackagePointer> packageHashTable = PackageHashTable
+						.fromJ9HashTable(classLoaderPointer.packageHashTable());
+				SlotIterator<J9PackagePointer> slotIterator = packageHashTable.iterator();
+				while (slotIterator.hasNext()) {
+					J9PackagePointer packagePtr = slotIterator.next();
+					if (filter.filter(packagePtr, filterArg)) {
+						outputter.print(packagePtr, out);
+					}
+				}
+			}
+		}
+	}
+
+
+	/**
+	 * Traverses through loaded classes in a module.
+	 * Uses outtputter to print details about classes
+	 * matched by filter and owned by the provided
+	 * module.
+	 * 
+	 * @param    out            The printstream that will be provided to outputter
+	 * @param    filter         Used to determine whether a class will be output.
+	 *                          Will be run on all loaaded classes owned by the module.
+	 * @param    outputter      Used to output details about a class that was matched
+	 *                          by the filter. Will be run on any class that filter
+	 *                          returns true for.
+	 * @param    filterArg      A string that is to be used as additional data for filter.
+	 *                          Will be passed to filter everytime filter is called.
+	 * @param    moduleAddress  The hex address of the module which will have it's classes
+	 *                          traversed.
+	 */
+	public static void iterateModuleClasses(PrintStream out, ClassIteratorFilter filter, ClassOutput outputter, String filterArg) throws CorruptDataException {
+		J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
```

Comment:
Instead of iterating through all the classes, you can iterate through the classes owned by the classloader which owns the module

Reply:
This is an issue with #3072: Moving to there